search/base.py

```diff
@@ -0,0 +1,18 @@
+"""
```

Comment:
there are many suggestions for structuring a python project and placement of unit tests, and [a](http://docs.python-guide.org/en/latest/writing/structure/#sample-repository) [few](http://learnpythonthehardway.org/book/ex46.html) of them contradict the structure of this codebase. i don't really agree with using code review convenience as a guide to structuring a project, and i can't say that i've ever found it really beneficial to have the two files next to each other, though others' results may vary. on top of that, if each project folder had a tests/ directory (where necessary), the file with the actual logic and the test file would be reasonably proximal anyway.

given the structure of the project elsewhere, i'm ok with leaving this as-is, but i think this is a pretty good illustration of the downfalls of mixing our test modules in with 'real' modules. we're putting a test class into a module whose name does not suggest that it has anything to do with tests. we might have other functionality (perhaps the ESConnection class i suggested elsewhere in the PR) that makes sense in `search/base.py`. then we have 'real' code mixed in with test 'code'. not the end of the world, but i think we should be avoiding that